src/React.Sample.Mvc4/Controllers/HomeController.cs

```diff
@@ -86,10 +88,25 @@ public ActionResult Comments(int page)
 		    var comments = _comments.Skip((page - 1) * COMMENTS_PER_PAGE).Take(COMMENTS_PER_PAGE);
 			var hasMore = page * COMMENTS_PER_PAGE < _comments.Count;
 
-		    return Json(new {
-				comments = comments,
-				hasMore = hasMore
-			}, JsonRequestBehavior.AllowGet);
+	        if (ControllerContext.HttpContext.Request.ContentType == "application-json")
```

Comment:
thanks
i'm going to add the output cache option as having 2 endpoints feels like duplication to me. and the code that generates the loadMoreClicked uses the Url of the link, so the JSX would have to be 'fudged' to make it work
mvc 6 is going to have better content negotiation built in